lib/astarte/client/appengine.ex

```diff
@@ -0,0 +1,52 @@
+#
+# This file is part of Astarte.
+#
+# Copyright 2021 SECO Mind
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule Astarte.Client.AppEngine do
+  @enforce_keys [:http_client]
+  defstruct @enforce_keys
+
+  @type t :: %__MODULE__{
+          http_client: Tesla.Client.t()
+        }
+
+  alias Astarte.Client.AppEngine
+  alias Astarte.Client.Credentials
+
+  @jwt_expiry 5 * 60
+
+  @spec new(String.t(), String.t(), String.t()) :: {:ok, t()} | {:error, any}
+  def new(base_api_url, private_key, realm_name) do
```

Comment:
Invert the order of `realm_name` and `private_key` (it makes more sense in terms of hierarchy imho)